larpandoracontent/LArHelpers/LArHierarchyHelper.cc

```diff
@@ -0,0 +1,1171 @@
+{
+    std::string str(prefix + "PDG: " + std::to_string(m_pdg) + " Energy: " + std::to_string(m_mainParticle ? m_mainParticle->GetEnergy() : 0) +
+        " Hits: " + std::to_string(m_caloHits.size()) + "\n");
+    for (const Node *pChild : m_children)
+        str += pChild->ToString(prefix + "   ");
+
+    return str;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+LArHierarchyHelper::MCHierarchy::ReconstructabilityCriteria::ReconstructabilityCriteria() :
+    m_minHits{15},
+    m_minHitsForGoodView{5},
+    m_minGoodViews{2},
+    m_removeNeutrons{true}
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+LArHierarchyHelper::MCHierarchy::ReconstructabilityCriteria::ReconstructabilityCriteria(const ReconstructabilityCriteria &obj) :
+    m_minHits{obj.m_minHits},
+    m_minHitsForGoodView{obj.m_minHitsForGoodView},
+    m_minGoodViews{obj.m_minGoodViews},
+    m_removeNeutrons{obj.m_removeNeutrons}
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+LArHierarchyHelper::MCHierarchy::ReconstructabilityCriteria::ReconstructabilityCriteria(const unsigned int minHits,
+    const unsigned int minHitsForGoodView, const unsigned int minGoodViews, const bool removeNeutrons) :
+    m_minHits{minHits},
+    m_minHitsForGoodView{minHitsForGoodView},
+    m_minGoodViews{minGoodViews},
+    m_removeNeutrons{removeNeutrons}
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+LArHierarchyHelper::RecoHierarchy::RecoHierarchy() :
+    m_pNeutrino{nullptr}
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+LArHierarchyHelper::RecoHierarchy::~RecoHierarchy()
+{
+    for (const Node *pNode : m_rootNodes)
+        delete pNode;
+    m_rootNodes.clear();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void LArHierarchyHelper::RecoHierarchy::FillHierarchy(const PfoList &pfoList, const bool foldToPrimaries, const bool foldToLeadingShowers)
+{
+    if (foldToPrimaries && !foldToLeadingShowers)
+    {
+        PfoSet primaries;
+        m_pNeutrino = LArHierarchyHelper::GetRecoPrimaries(pfoList, primaries);
```

Comment:
Can move these two lines out of the if statement as they are common to all cases?